controllers/search_controller/mongodbsearch_reconcile_helper.go

```diff
@@ -264,7 +270,32 @@ func (r *MongoDBSearchReconcileHelper) ValidateSingleMongoDBSearchForSearchSourc
 		for i, search := range searchList.Items {
 			resourceNames[i] = search.Name
 		}
-		return xerrors.Errorf("Found multiple MongoDBSearch resources for search source '%s': %s", r.db.Name(), strings.Join(resourceNames, ", "))
+		return xerrors.Errorf(
+			"Found multiple MongoDBSearch resources for search source '%s': %s", r.db.Name(),
+			strings.Join(resourceNames, ", "),
+		)
+	}
+
+	return nil
+}
+
+func (r *MongoDBSearchReconcileHelper) ValidateSearchImageVersion() error {
+	version := strings.TrimSpace(r.mdbSearch.Spec.Version)
+	if version != "" {
```

Comment:
If the version field in the spec is empty, you should check `r.operatorSearchConfig.SearchVersion` (see the `createOrUpdateStatefulSet()` method on `MongoDBSearchReconcileHelper` and perhaps extract a helper to calculate the version).